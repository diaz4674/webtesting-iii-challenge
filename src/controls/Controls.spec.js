// Test away
import React from 'react';
import renderer from 'react-test-renderer'; // 1: install this npm module as a dev dependency

import {render, cleanup,fireEvent} from '@testing-library/react'
import '@testing-library/react/cleanup-after-each'
import Controls  from './Controls'


afterEach(cleanup)

describe('<Controls/>', () => {
    it('matches snapshot', () => {
        const tree = renderer.create(<Controls/>)
        expect(tree.toJSON()).toMatchSnapshot()
    })

    it('provided button to toggle closed and locked states', () => {
        const gateLock = render(<Controls />)
        expect(gateLock.getByTestId(/toggle-locked/i))
        expect(gateLock.getByTestId(/toggle-closed/i))
    })

    it('toggle-locked renders "lock gate"when unlocked, and "close gate" when gate is open', () => {
        const gate = render(<Controls locked={false}/>)
        expect(gate.getByText(/lock gate/i))
        expect(gate.getByText(/close gate/i))
    })

    it('toggle-closed renders "unlock gate" and "open gate"  when gate is locked/closed', () => {
        const gate = render(<Controls locked closed/>)
        expect(gate.getByText(/unlock gate/i))
        expect(gate.getByText(/open gate/i))
    })

    it('when gate is locked/open, toggle-closed is disabled', () => {
        const gate = render(<Controls locked/>)
        const closedButton = gate.getByTestId(/toggle-closed/i)
        const lockedButton = gate.getByTestId(/toggle-locked/i)
        expect(closedButton.disabled).toBe(true)
        expect(lockedButton.disabled).toBe(true)
    })

})