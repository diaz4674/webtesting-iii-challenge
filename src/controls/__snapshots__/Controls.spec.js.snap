// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls/> matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="toggle-locked"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="toggle-closed "
  >
    Close Gate
  </button>
</div>
`;
